includes/class-wc-stripe-webhook-handler.php

```
@@ -1827,6 +1844,157 @@ protected function handle_deferred_payment_intent_succeeded( $order, $intent_id
+		//
+		// For a captured charge, refund the charge's remaining balance (amount minus any prior
+		// refund) rather than the order total, which can drift from what Stripe actually holds after
+		// a partial capture, an earlier partial refund, or an edited order total. process_refund()
+		// treats a null amount as a no-op for a captured charge, so a real amount is required; the
+		// authorisation-void path instead needs null (any amount makes it throw).
+		$refund_amount = null;
+		if ( ! $is_authorization ) {
+			// Missing or delayed charge data: don't guess an amount and risk over-refunding. Leave the
+			// order unflagged so an Action Scheduler retry can settle it once the charge is available.
+			if ( ! is_object( $charge ) || ! isset( $charge->amount, $charge->currency ) ) {
+				WC_Stripe_Logger::warning( "Skipped refunding cancelled order {$order->get_id()} for Stripe PaymentIntent {$intent_id} - no charge amount available to refund." );
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'WC_Stripe_Logger' in method 'handle_deferred_payment_for_cancelled_order'. (undefined)

(StaticAccess)